### ISSUE TITLE
Rebooting on LinOS 14

### ISSUE THREAD

**Issue by mfascioglu** [janithcooray/lin_os_swap_mod#27]
Swap creation complated and rebooted then swap is activated on boot. Device is opening successful but when writing some amount data on swap device is rebooting. I checked with [Ram Truth](https://play.google.com/store/apps/details?id=sa.ramtruth&hl=tr) app. Do you have any idea for this problem?

Device: Xperia XZ1
Rom: https://xdaforums.com/t/14-rom-lineageos-21-0-unofficial-1-0-2024-02-25.4659246/

**Comment by janithcooray** [comment]
Hi, Could you elaborate on what you meant by "writing some amount data". were you trying to open some apps assumeing it would get added to swap?

**Comment by mfascioglu** [comment]
Thanks for the reply.

Yes, Swap creation and swap activation is successfully. when device uses created swap file, device is rebooting. (When 1-2 MB of a 4 GB swap file is used.)

**Comment by janithcooray** [comment]
@yeezylife check if it's actually working or if its on fail safe mode

**Comment by janithcooray** [comment]
@yeezylife 

go to adb shell as root and go ```sh
cat /data/swap/swapfile.log
```
this should show what happened. if its failing , then there's an incompatibility the version of the swapon bin or it's non-existent at all. will you be able to contact me via email or telegram? it'll be handy if can send you custom configured zips to figure this out.

**Comment by janithcooray** [comment]
It's failing as it doesnt show the part where swap starts. i'll try adding more log data to the file so we can find out whats wrong. i'll link a debug release

**Comment by janithcooray** [comment]
@yeezylife check https://github.com/janithcooray/lin_os_swap_mod/releases/tag/v2.0-c-debug-1
be careful, this is a debug version. uninstall once you get the /data/swap/swapfile.log

**Comment by janithcooray** [comment]
@yeezylife  right, swapon file may be somewhere else, could you send me the link from where you downloaded this ROM

**Comment by janithcooray** [comment]
@yeezylife https://github.com/janithcooray/lin_os_swap_mod/releases/tag/v2.0-c-debug-2
see what happens with this one, it might actually start swap

**Comment by janithcooray** [comment]
@yeezylife thanks for the help, i will need some more assistnace to figure this out.

The swap bin is in the default path. not sure why it's not working at this point. Could you try this release.
https://github.com/janithcooray/lin_os_swap_mod/releases/tag/v2.0-c-debug-3

Also could you see the output of the following on adb

```sh
swapon -v
```
or
```sh
swapon --version
```
and
```sh
ls -lah /data/swap/swapfile
```

**Comment by janithcooray** [comment]
@yeezylife  thanks for your help. i will look for another way to debug this in the meantime. Sorry that this didnt work out for you

**Comment by janithcooray** [comment]
Marking as closed since fail safe will prevent reboot. It will instead disable the module function. If you encounter this issue please post on #33
